### PR TITLE
Edits to day 1 content

### PR DESCRIPTION
**Pull request by ecraig12345**
- Mostly minor copyediting and formatting
- Switch all == to === (want to model good practices)
- Add links to more info about `this`
- Add links to MDN docs for some terms

### CODE REVIEW

**Comment by ecraig12345** [comment]
Realized I forgot to finish updating the description of controlled vs uncontrolled in step 6. I'll close this until that's finished.

File: step1-04/demo/README.md [hunk 1]
```diff
@@ -160,39 +159,37 @@ Now that we've got two Counters on our page, we can see that the string passed i
...
 Our next step is to wire up the button to increment the `counter` in our component state. This will very similar to what we did in step 3, but instead of placing the function in a script tag, we can create it as a class method, and keep it out of the global scope.
 
-> By convention we place methods below render, and private methods (those for internal use only) are prefixed with an underscore.
+> By convention we place other methods below render, and private methods (those for internal use only) are prefixed with an underscore.
+
+This function will update our component's state, incrementing the counter value by 1. (Note that `setState` only modifies the values of keys listed in object passed as its parameter.)
 
 ```jsx
 _onButtonClick = () => {
-  this.setState(prevState => ({ counter: prevState.counter + 1 }));
```
Comment (micahgodbolt) [inline step1-04/demo/README.md]: this change might have crept in while you were doing edits. The functional state is prefered here.

Comment (micahgodbolt) [reply step1-04/demo/README.md]: please revert this change as well

**Comment by micahgodbolt** [comment]
Also, BRAVO! Thanks for all the fine tooth combing you've done.

**Comment by ecraig12345** [comment]
Reopening. I'll submit the controlled/uncontrolled thing separately...

**Comment by ecraig12345** [comment]
Huh I can't reopen for some reason... Made this one instead: https://github.com/Microsoft/frontend-bootcamp/pull/32

**Pull request closed without merging**